### ISSUE TITLE
Thank you and looking forward to seeing the variational auto-encoder topic

### ISSUE THREAD

**Issue by Khoa-NT** [naokishibuya/deep-learning#2]
Dear Sir,

I'm really sorry. I don't know how to contact you via email so I write this on your GitHub. 

How are you? I haven't seen your new post on Medium. I hope you are doing well.
I just read your tutorials on Medium ( Entropy, Cross-Entropy, KL, Deconvolutional..)
Thank you so much for spending time to write great tutorials. Now I can understand about entropy clearly.

I hope and I'm looking forward to seeing the tutorial about variational auto-encoder as you said in the KL tutorial.

Thank you again.
Sincerely,
Khoa

**Comment by naokishibuya** [comment]
@shaolinkhoa thanks for asking about VAE article.

I've been planning for VAE article for quite some time and I know it will be a series of articles.  At this moment, however, I don't have any concrete timing when I can publish them to Medium.

For the time being, I'd recommend the following video tutorial for a quick overview.

https://www.youtube.com/watch?v=9zKuYvjFFS8

For more technical details, I'd recommend the following tutorial paper on VAE.

https://arxiv.org/pdf/1606.05908.pdf
 
Hope that helps.

**Comment by Khoa-NT** [comment]
Thank you for your quick reply.
Because I'm reading the paper about VAE and they use KL in VAE just like you said
"The KL divergence is used to force the distribution of latent variables to be a normal distribution so that we can sample latent variables from the normal distribution. As such, the KL divergence is included in the loss function to improve the similarity between the distribution of latent variables and the normal distribution."
So that made me curious about KL in VAE.

**Comment by Khoa-NT** [comment]
Dear Sir,
I read the "Tutorial on Variational Autoencoders". 
But I can't understand the part  **2.1 Setting up the objective** : the equation (3)

![Equation three](http://latex.codecogs.com/png.latex?%5Cdpi%7B150%7D%20%5Csmall%20D_%7BKL%7D%5BQ%28z%20%5Cvert%20X%29%20%5Cvert%5Cvert%20P%28z%20%5Cvert%20X%29%5D%20%26%3D%20E_%7Bz%5Csim%20Q%28z%20%5Cvert%20X%29%7D%5B%5Clog%20Q%28z%20%5Cvert%20X%29%20-%20%5Clog%20P%28X%20%5Cvert%20z%29%20-%20%5Clog%20P%28z%29%5D%20&plus;%20E_%7Bz%5Csim%20Q%28z%20%5Cvert%20X%29%7D%5B%5Clog%20P%28X%29%5D)

`Here, log P(X) comes out of the expectation because it does not depend on z.`
It made me confusing. My friend told me that we can assume logP(x) as a constant in this expectation equation and then let it out. 
But follow your KL tutorial, I think It should be: 
![the second equation](http://latex.codecogs.com/png.latex?%5Cdpi%7B150%7D%20%5Csmall%20E_%7Bz%5Csim%20Q%28z%20%5Cvert%20X%29%7D%5B%5Clog%20P%28X%29%5D%20%3D%20%5Csum_%7Bz%7DQ%28z%20%5Cvert%20X%29%5Cast%20%5Clog%20P%28X%29)

But in the end, we only have logP(x). Does it mean we have the sum equal 1?
![Q](http://latex.codecogs.com/png.latex?%5Cdpi%7B150%7D%20%5Csmall%20%5Csum_%7Bz%7DQ%28z%20%5Cvert%20X%29%3D%201)

**Comment by carlee0** [comment]
@shaolinkhoa, I think your argument is exactly correct. In your second equation you could see that X is given in the distribution Q(z|X) and P(X) does not depend on z, therefore p(X) is a constant in this summation. In the last equation the summation should indeed be evaluated to 1 if it is a discrete distribution.
